sci-mathematics/octave: improve metadata

fkobi opened the pull request:
<!-- Please put the pull request description above -->

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

thesamesam commented:
If bzip2 is linked against, please keep it in `*DEPEND`.

thesamesam commented:
https://github.com/gnu-octave/octave/blob/fe1a26d2e2c62e3c7058a0b1348d4da9a544d5d3/configure.ac#L1527

orlitzky commented:
> In that case it deserves it's own USE flag I feel.

You would be right, but the upstream build system doesn't have `--disable-bzip2`, so the only safe option is to require it unconditionally. Otherwise octave could "accidentally" be build linked to bzip2, and then if bzip2 is removed, octave will crash.

Octave 11.1.0 still has the same behavior, if you really want to make the dependency go away, the best thing to do would be to add that `--disable-bzip2` flag upstream and _then_ put it behind a USE flag.

fkobi committed:
sci-mathematics/octave: metadata: complete upstream

Signed-off-by: Filip Kobierski <fkobi@pm.me>

fkobi committed:
sci-mathematics/octave: update HOMEPAGE & LICENSE

For the license change:
- GPL-3+: src/main-cli.cc#L13
- CC0: libgui/src/icons/license.md?plain=1#L35
- public-domain: libgui/src/icons/license.md?plain=1#L10
- conditional: libgui/languages/build_ts/octave-qt/qpagesetupdialog.cpp#L2

Signed-off-by: Filip Kobierski <fkobi@pm.me>
